libraries/AP_Mount/AP_Mount_Siyi.cpp

```diff
@@ -699,9 +699,16 @@ float AP_Mount_Siyi::get_zoom_mult_max() const
 // set zoom specified as a rate or percentage
 bool AP_Mount_Siyi::set_zoom(ZoomType zoom_type, float zoom_value)
 {
+    // rate zoom (continuous)
     if (zoom_type == ZoomType::RATE) {
-        // disable absolute zoom target
-        _zoom_mult_target = 0;
+        if (zoom_value > 0) {
```

Comment:
I'd prefer we keep the handling of zoom rate control and zoom position control separate.  The reason is that rate control is a lower level controller and is thus more reliable.  The absolute zoom control relies on feedback from the camera.. and this feedback could fail.

So I think instead we should record the last received zoom_type and zoom_value and then in update_zoom_control we should add support for re-sending the zoom_rate at regular intervals.  Once the user has set the zoom rate to zero though we should stop sending the zoom-rate command to the camera because it appears to me that setting zoom-rate=0 uses the MANUAL_ZOOM_AND_AUTO_FOCUS command which may enable auto focus... so if we kept sending zoom-rate=0 forever the user could never manually focus the camera.